### ISSUE TITLE
Android Mattermost App not connecting to Server

### ISSUE THREAD

**Issue by phantomofrussia** [mattermost/mattermost-server#7440]
After putting in my server url, I get the following message "Cannot connect to the server. Please check your server URL and internet connection."

Uninstalled App reinstalled and happens every time. On coworkers iPhones, this does not happen and they can connect just fine. Using latest version of the App.

I know I have internet connection on my phone and if I go to the url using the phone web browser I get the login page. 

The desktop app, ios app, and web gui all work without any problems. 

Any help would be appreciated!

Thanks,
Simon

**Comment by lindy65** [comment]
Hello @phantomofrussia,

Thanks for your feedback,

Could you share which server version of Mattermost you're using as well as an example of the URL you're trying to connect to?

Sometimes using `https://` vs `http://` can make a difference and the url shouldn't include the team name - just the server.  e.g. `https://servername.domainname.com`

Also are you using the Mattermost Classic app or the react native Mattermost Beta app?

**Comment by lindy65** [comment]
Hi @phantomofrussia 

Circling back on this - is your issue resolved?

**Comment by phantomofrussia** [comment]
Hi @lindy65,

Sorry for not responding back to you sooner. This is not resolved unfortunately. 

Server version is 4.1 and the url is chat.infostructures.com

I have tried both https and http in the url on the app but it is not working. Getting the same error as before. I am using the new Mattermost App, have not tried the older one.

**Comment by lindy65** [comment]
Hi @phantomofrussia 

Thanks for your feedback,

You could try upgrading to the [latest server version](https://about.mattermost.com/download/) which released on September 16th.

Perhaps also check in the system console what your `Site URL` is set to?  System Console > Settings > General > Configuration > Site URL

Let me know if this helps?

**Comment by phantomofrussia** [comment]
Hi @lindy65,

I can try upgrading to the latest version. 
I want to point out that I am only having this issue with the Android App. The iOS app and desktop app work just fine so I do not think there is a configuration issue on the server end....

**Comment by lindy65** [comment]
Thanks @phantomofrussia,

I'll post your issue to the mobile app team and ask for their help...

**Comment by MusikPolice** [comment]
@phantomofrussia is it possible for you to post some more details about your configuration? I'd like to know the version information of the android app that is giving you trouble, the exact version of your server, and see a screenshot of the System Console > Settings > General > Configuration screen from your server, as well as the relevant configuration screen from your mobile app.

**Comment by enahum** [comment]
Looks to me that this can be an issue cause by SSL, maybe an unsupported CA or unsupported Cipher, @phantomofrussia have you tried connecting to `https://pre-release.mattermost.com` if so do you get the same error or can you reach the login screen?

**Comment by Beachwhale13** [comment]
I am having the same problems as @phantomofrussia. Works on my desktop, not on my samsung phone. I can connect to the pre-release.mattermost.com url listed above, but I get a "Mattermost server unreachable, incompatible or SSL is not properly configured" message when going to the url I need to be at.

**Comment by enahum** [comment]
Then it might be related to the SSL certificate, @beachwhale13 what happens if you try and open the url with the mobile browser?

Are you using nginx in front as reverse proxy? If yes can you share the config file?

**Comment by Beachwhale13** [comment]
I can open the link on my mobile browser. I'm new to the whole github mattermost thing I don't think I"m using nginx. Just downloaded the app to my android phone and tried to start up mattermost.

**Comment by phantomofrussia** [comment]
@enahum, I am able to get to the login page for pre-release.mattermost.com but not my server URL. I am not getting the error that beachwhale13 is getting though. 

I did a ssl check on ssllabs.com and it is saying my certificate chain is incomplete. I can work on fixing that, just really surprised how an iOS device is not complaining about it but an Android device is. I have seen this the other way around.

**Comment by phantomofrussia** [comment]
Also my error is: "Cannot connect to the server. Please check your server URL and internet connection"

**Comment by enahum** [comment]
@phantomofrussia what you describe is basically an SSL issue, please try and correct your SSL configuration and see if that solves the problem

**Comment by andreasoc** [comment]
i have the same problem
i'm locally with the bitnami ova with the last mattermost version
go in android in wifi in my lan
insert the local site http://172.16.X.X and dont' work.
in the System Console > Settings > General > Configuration > Site URL there is the same link

thanks guys!

**Comment by cedeon** [comment]
I have the same problem too.  I'm trialing this for potential to go to enterprise but its imperative I have a mobile working solution.  I'm using the standard docker deployment and a proper SSL cert.  Web works fine but on android the error is same as @phantomofrussia

**Comment by lindy65** [comment]
Thanks @andreasoc @cedeon 

Do you have any ideas @enahum ?

**Comment by enahum** [comment]
Well what I would do is:
1. Configure the server to accept HTTP requests, if the connection is successful then try with SSL enabled.
2. Check that the SSL is properly installed and configured meaning not just the certificate but any intermediary certificate that might be needed, ensure that is not a self-sign certificate, also review the ciphers that are being used and see if they are actually supported in the devices.
3. Try and establish a direct connection to the server without the use of a reverse proxy to see if that helps (the problem might be in the reverse proxy itself)
4. Check logs everywhere, check the logs in nginx/apache/etc.. the mattermost logs and if possible device logs (like logcat) to see if there is something that might be helpful to understand the issue.

nothing else comes to mind right now but I believe the previous points can be helpful

**Comment by lindy65** [comment]
Hi @cedeon @andreasoc @phantomofrussia 

Just circling back to find out whether you have resolved your issue?

**Comment by willruss1** [comment]
I'm having the same issue. Web UI, desktop app, and even mobile web work fine on a newly created MM instance (based on the production-docker configuration), all over SSL with no discernible errors.

What doesn't work, is the Android app. All I or anyone gets is "Cannot connect to the server. Please check your server URL and internet connection" though the exact same URL works just fine for the web UI on the same devices.

**Comment by enahum** [comment]
@willruss1 can you share the url of your Mattermost instance? Normally this is a symptom of a problem with SSL maybe an intermediate certificate.

**Comment by willruss1** [comment]
@enahum, sure can. https://chat.digi-merc.org, though I found the problem based on your suggestion.

Turns out the cert chain was in the wrong order. That's what I get for building things hastily at 2am, I guess. A quick check on ssl-decoder.org pointed out the issue, I should have caught it before. Odd that no browser complained about that, honestly.

Sorry to have bothered you, but maybe this will help others avoid the same oversight.
